apps/api/v2/src/modules/oauth/guard/oauth-client/oauth-client.guard.ts

```diff
@@ -0,0 +1,29 @@
+import { OAuthClientRepository } from "@/modules/oauth/oauth-client.repository";
```

Comment:
I tried placing this guard in "modules/auth/guard" but ran into circular dependency issue - this "OAuthClientGuard" guard requires "OAuthClientRepository". If we moved the guard to "AuthModule", then we need to import the "OAuthClientModule" that has the repository to inject in the guard, but the "OAuthClientModule" already imports "AuthModule".

Reply:
I created a branch of this branch to continue the work, and now tried importing the new guard into the "user.module.ts" to use the guard on user controller, and run into circular dependency again:

user.module imports OAuthClientModule to use the guard, but OAuthClientModule already imports user.module :/

Reply:
does it make sense to import whole module when we only need the repository for example ?

Reply:
maybe it makes sense to declare oAuth as a global module since most of the endpoints will use it

Reply:
Made oAuth client module with its new guard global and made it export oauth repository and the new guard - now other modules like user module can simply use @UseGuards(OAuthClientGuard).

File: https://github.com/calcom/cal.com/pull/12814/files#diff-b17562a14071a985176d16db25acbe2c3fdf94013c3fd68a08923ed536ca25f6